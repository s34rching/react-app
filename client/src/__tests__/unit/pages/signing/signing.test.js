import React from 'react';
import { shallow } from 'enzyme';
import { SigningPage, mapStateToProps } from '../../../../pages/signing/signing.component';
import getMockedState from '../../../utils/mock-state-provider';

describe('Pages: Signing Page', () => {
  const error = { message: 'something bad happened' };

  describe('Signing Page', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<SigningPage />);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    it('should render without errors', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render error block', () => {
      wrapper.setProps({ error });

      expect(wrapper.find('ErrorContainer').text()).toEqual(error.message);
    });
  });

  describe('mapStateToProps', () => {
    it('should map state', () => {
      const initialState = getMockedState(['user']);
      initialState.user.error = error;

      expect(mapStateToProps(initialState)).toContainEntry(['error', initialState.user.error]);
    });
  });
});
