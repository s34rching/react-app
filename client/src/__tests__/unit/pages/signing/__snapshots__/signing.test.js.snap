// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages: Signing Page Signing Page should render without errors 1`] = `
<SigningPageContainer>
  <SigningPageFormsContainer>
    <Connect(SignIn) />
    <Connect(SignUp) />
  </SigningPageFormsContainer>
</SigningPageContainer>
`;
